firmware/main.c

```diff
@@ -60,343 +101,539 @@ PROGMEM const uint8_t configurationReply[6] = {
   MICRONUCLEUS_WRITE_SLEEP,
   SIGNATURE_1,
   SIGNATURE_2
-};  
+};
 
-  typedef union {
+typedef union {
     uint16_t w;
     uint8_t b[2];
-  } uint16_union_t;
-  
+} uint16_union_t;
+
 #if OSCCAL_RESTORE_DEFAULT
   register uint8_t      osccal_default  asm("r2");
-#endif 
+#endif
 
-register uint16_union_t currentAddress  asm("r4");  // r4/r5 current progmem address, used for erasing and writing 
-register uint16_union_t idlePolls       asm("r6");  // r6/r7 idlecounter
+register uint16_union_t currentAddress asm("r4");  // r4/r5 current progmem address, used for erasing and writing
+register uint16_union_t idlePolls asm("r6");  // r6/r7 idle counter - each tick is 5 milliseconds
 
-// command system schedules functions to run in the main loop
+// command used to trigger functions to run in the main loop
 enum {
-  cmd_local_nop=0, 
-  cmd_device_info=0,
-  cmd_transfer_page=1,
-  cmd_erase_application=2,
-  cmd_write_data=3,
-  cmd_exit=4,
-  cmd_write_page=64  // internal commands start at 64
+    cmd_local_nop = 0,
+    cmd_device_info = 0,
+    cmd_transfer_page = 1,
+    cmd_erase_application = 2,
+    cmd_write_data = 3,
+    cmd_exit = 4,
+    cmd_write_page = 64  // internal commands start at 64
 };
-register uint8_t        command         asm("r3");  // bind command to r3 
-
-// Definition of sei and cli without memory barrier keyword to prevent reloading of memory variables
```

Comment:
this actually helped to shave off many bytes in older versions of gcc. Is it not needed anymore?

Reply:
Moved to line 128 after the command enum

Reply:
oh, i meant the defines. But I noted you just inlined them. should be fine,